Modify post processing nnunet for CSA computation and QC report (#57)
sct-pipeline/contrast-agnostic-softseg-spinalcord: README.md

Specify the path of preprocessed dataset with the flag `-path-data`. 

### 3.1.Launch processing

```
sct_run_batch -jobs -1 -path-data <PATH_DATA> -path-output <PATH-OUTPUT> -script process_data.sh -script-args exclude.yml
```

A `process_data_clean` folder is created in <PATH-OUTPUT> where the cropped data and derivatives are included. Here, only the images that have a manual segmentation and soft segmentation are transfered.

### 3.2.Quality control

> After running the analysis, check your Quality Control (qc) report by opening the file <PATH-OUTPUT>/qc/index.html. Use the "search" feature of the QC report to quickly jump to segmentations or labeling issues.

Reviewer: ```suggestion
After running the analysis, check your Quality Control (QC) report by opening the file <PATH-OUTPUT>/qc/index.html. Use the "search" feature of the QC report to quickly jump to segmentations or labeling issues.
```

``` 
## 6.Analyse CSA results
To generate violin plots and analyse results, put all CSA results file in the same folder (here `csa_nnunet_vs_ivadomed`) and run:

```
python analyse_csa_nnunet_ivadomed.py -i-folder ~/duke/projects/ivadomed/contrast-agnostic-seg/csa_measures_pred/csa_nnunet_vs_ivadomed/ \
                                      -include ivado_hard_GT ivado_avg_bin_no_crop ivado_soft_no_crop csa_nnunet_soft_avg_all_no_crop \
```
* `-i-folder`: Path to folder containing CSA results from models to analyse
* `-include`: names of the folder names to include in the analysis (one model = one foler)

### 6.1.Analyse CSA IVADOMED only
To generate violin plots to compare 4 ivadomed models (original):

```
python gen_charts.py --contrasts T1w T2w T2star rec-average_dwi \
       --predictions_folder ../duke/projects/ivadomed/contrast-agnostic-seg/csa_measures_pred/group8-9_combined-2022-12-21/ \
       --baseline_folder ../duke/projects/ivadomed/contrast-agnostic-seg/archive_derivatives_softsegs-seg/contrast-agnostic-centerofmass-preprocess-clean-all-2022-10-22\results_MTS_renamed
```
  

> ## 7.Run qc report on prediction masks from other datasets

Reviewer: ```suggestion
## 7. Run QC report on prediction masks from other datasets
```

~~~
sct_run_batch -path-data <PATH_DATA> -path-out <PATH-OUT> -script-args <PATH_PRED_MASK> -jobs 20 -script run_qc_prediction_XXX.sh
~~~
* `-path-data`: Path to the original dataset used to run inferences.
* `-path-output`: Path to the results folder to save QC report
* `-script`: Script `run_qc_prediction_XXX` corresponding to the dataset.
* `-script-args`: Path to prediction masks for the specific dataset
  
### 7.1.Example running QC on prediction masks from nnUnet from other datasets
~~~
sct_run_batch -jobs 20 -path-data ~/data_nvme_sebeda/datasets/dcm-zurich/ \
                       -path-output ~/data_nvme_sebeda/qc_dcm_zurich_sag_nnUnet_2023-05-30 \
                       -script run_qc_prediction_dcm_zurich_sag.sh \
                       -script-args ~/duke/temp/muena/contrast-agnostic/pure-inference/Dataset725_dcmZurichSagittalRPI/test713_softAvg/
~~~

## 8.Active learning procedure
To extend the training set to other contrasts and to pathologies, we applided the segmentation model to other datasets, manually corrected the segmentations and added them to the training set.

Here is the detailed procedure:

> 1. Run inference on other datasets for the selected models and QC report.

Reviewer: ```suggestion
1. Run inference on other datasets for the selected models and generate the QC report from prediction masks.
```

> 2. Select ~20 interesting images per datasets (in QC).

Reviewer: ```suggestion
2. Select ~20 interesting images per dataset (using the QC report).
```

> 3. Correct the inference on the selected subject if needed.

Reviewer: Will the inference be BIDS-compatible? If so, we can add info about `manual-correction`:
```suggestion
3. Correct the inference on the selected subjects if needed (you can use [`manual-correction`](https://github.com/spinalcordtoolbox/manual-correction) script).
```

> 4. Add the inferred segmentations to each dataset (under derivatives/labels_contrast_agnostic).

Reviewer: ```suggestion
4. Add the inferred segmentations to the `derivatives/labels_contrast_agnostic` folder of each dataset.
```
